rpmio/macro.c

```diff
@@ -1545,33 +1581,14 @@ expandMacro(MacroBuf mb, const char *src, size_t slen)
 	    continue;
 	}
 
-	/* Setup args for "%name " macros with opts */
+	/* Grab args for parametric macros */
 	ARGV_t args = NULL;
 	if (me->opts != NULL) {
 	    argvAdd(&args, me->name);
 	    if (lastc)
 		se = grabArgs(mb, &args, fe, lastc);
 	}
-
-	rpmMacroEntry prevme = mb->me;
-	ARGV_t prevarg = mb->args;
-	if (args != NULL)
-	    setupArgs(mb, me, args);
-
-	/* Recursively expand body of macro */
-	if (me->body && *me->body) {
-	    if ((me->flags & ME_LITERAL) != 0)
-		mbAppendStr(mb, me->body);
-	    else
-		expandMacro(mb, me->body, 0);
-	}
-
-	/* Free args for "%name " macros with opts */
-	if (args != NULL)
-	    freeArgs(mb);
-	mb->args = prevarg;
-	mb->me = prevme;
-
+	doExpandThisMacro(mb, me, args);
```

Comment:
Oh, the other thing is that this looks a lot like a memory leak, as the argv free is buried far away from here. I'd prefer symmetry here - args was initially allocated here so it should also be freed here, ditto with optargs in expandThisMacro().